tests/test_transformations.py

```diff
@@ -0,0 +1,177 @@
+                                                    strategy='maximum')
+
+    with h5py.File(video_fixture, 'r') as open_video:
+        video = open_video['data'][:]
+        match_count = 0
+        new_video_index = 0
+        sampling_ratio = int(input_fps / output_fps)
+        expected_count = math.ceil(len(video) / int((input_fps / output_fps)))
+
+        for i in range(0, len(video), sampling_ratio):
+            frames = video[i:(i + sampling_ratio)]
+            new_frame = new_video[new_video_index]
+            test_frame = new_frame - np.max(frames, axis=0)
+            if not test_frame.any():
+                match_count += 1
+            new_video_index += 1
+        assert expected_count == match_count
+
+
+def test_video_average_downsampling(video_fixture):
+    input_fps = 31
+    output_fps = 4
+    new_video = transformations.downsample_h5_video(video_path=video_fixture,
+                                                    input_fps=input_fps,
+                                                    output_fps=output_fps,
+                                                    strategy='average')
+
+    with h5py.File(video_fixture, 'r') as open_video:
+        video = open_video['data'][:]
+        match_count = 0
+        new_video_index = 0
+        sampling_ratio = int(input_fps / output_fps)
+        expected_count = math.ceil(len(video) / int((input_fps / output_fps)))
+
+        for i in range(0, len(video), sampling_ratio):
+            frames = video[i:(i + sampling_ratio)]
+            new_frame = new_video[new_video_index]
+            test_frame = new_frame - np.mean(frames)
+            if not test_frame.any():
+                match_count += 1
+            new_video_index += 1
+        assert expected_count == match_count
+
+
+@pytest.mark.parametrize(("coordinate_pair, window_size, video_shape,"
+                          "expected_pair"), [
+                        ((0, 9), (10, 10), (512, 512), (4, 9)),
+                        ((9, 0), (10, 10), (512, 512), (9, 4)),
+                        ((511, 4), (10, 10), (512, 512), (506, 4)),
+                        ((4, 511), (10, 10), (512, 512), (4, 506))])
+def test_coordinate_window_shift(coordinate_pair, window_size, video_shape,
+                                 expected_pair):
+    transformed_center = transformations.get_transformed_center(coordinate_pair,
+                                                                     window_size,
+                                                                     video_shape)
+    assert transformed_center == expected_pair
+
+
+@pytest.mark.parametrize(("coordinate_pair, window_size"),
+                         [((10, 10), (6, 6)),
+                          ((10, 10), (12, 12)),
+                          ((510, 510), (10, 10)),
+                          ((10, 510), (10, 10)),
+                          ((510, 10), (10, 10))])
+def test_video_subset_coord_pair(coordinate_pair, window_size):
```

Comment:
Testing this would be a lot easier if the coordinate extraction was separated from the video writing in the function, don't you think?

Reply:
I think the math is trivial enough its not too cumbersome, it would only be four lines to get the column and row information.